### DISCUSSION TITLE
IWP TELEPORT

### DISCUSSION THREAD

**Discussion by cho66x** [naverz/zepeto-studio-global#41] [General]
Hello Zepetto-Tech Evangelist, How to make payments in the world. if the player wants to teleport, then he has to pay. I can't think of any logical code in the client server. please tell me about this,

**Comment by cho66x** [comment]
or like this, how to make a paid event ui in my world

**Comment by naverz-mkao** [comment]
Hi Anggacho,

We have an IWP (In World Purchase) system that you can use!

https://studio.zepeto.me/guides/in-world-purchases

<img width="500" alt="image" src="https://user-images.githubusercontent.com/107931355/184631340-24af0bba-ac49-4c74-b343-4f8db8126520.png">

You can run your player teleport code inside the OnPurchaseComplete callback that Zepeto provides.

https://studio.zepeto.me/guides/linking-purchasecomplete-event1

<img width="500" alt="image" src="https://user-images.githubusercontent.com/107931355/184632677-9057e06d-6eb6-46bb-85b7-26b920bed5dd.png">



What you can do is have a trigger volume around your teleport mesh. When the player enters that trigger volume, you can then make your button to buy a "Teleport Ticket" visible. When the player clicks on that button, the confirmation to buy the "Teleport Ticket" will pop up. 

<img width="143" alt="image" src="https://user-images.githubusercontent.com/107931355/184634028-cb984ff3-4fc9-4e46-b874-75224e2affac.png">


Once the player makes the purchase, then you can run your teleport code inside the OnPurchaseComplete callback.

Note that to set up purchase items, you will need to make a world product in the world console. 

https://studio.zepeto.me/guides/making-a-world-console
https://studio.zepeto.me/guides/making-world-products

Also make sure to register a World cover image, as you may run into errors with purchasing if you do not do this step.

<img width="653" alt="image" src="https://user-images.githubusercontent.com/107931355/184635113-faf9cd22-b917-42e3-b8af-5ecf26100589.png">

Let us know if this helps!

**Comment by cho66x** [comment]
well i will follow these steps first. but I am very happy with your answer, thank you very much for your answer.

**Reply by naverz-mkao** [reply]
Happy to help, let us know if you run into any issues!
